include/stencils.hxx

```diff
@@ -46,49 +46,73 @@ void inline populateStencil(stencil &s, const FieldType& f, const typename Field
   
   switch(stagger) {
   case(STAGGER::None):
-    if (nGuard == 2) s.mm = f[i.template minus<2, direction>()];
+    if (nGuard == 2) {
+      if (direction == DIRECTION::YOrthogonal) {
+        s.mm = f.ynext(-2)[i.template minus<2, direction>()];
+      } else {
+        s.mm = f[i.template minus<2, direction>()];
+      }
+    }
     if (direction == DIRECTION::YOrthogonal) {
-      s.m = f.ydown()[i.template minus<1, direction>()];
+      s.m = f.ynext(-1)[i.template minus<1, direction>()];
     } else {
       s.m = f[i.template minus<1, direction>()];
     }
     s.c = f[i];
     if (direction == DIRECTION::YOrthogonal) {
-      s.p = f.yup()[i.template plus<1, direction>()];
+      s.p = f.ynext(1)[i.template plus<1, direction>()];
     } else {
       s.p = f[i.template plus<1, direction>()];
     }
-    if (nGuard == 2) s.pp = f[i.template plus<2, direction>()];
+    if (nGuard == 2) {
+      if (direction == DIRECTION::YOrthogonal) {
+        s.pp = f.ynext(2)[i.template plus<2, direction>()];
+      } else {
+        s.pp = f[i.template plus<2, direction>()];
+      }
+    }
     break;
   case(STAGGER::C2L):
-    if (nGuard == 2) s.mm = f[i.template minus<2, direction>()];
+    if (nGuard == 2) {
+      if (direction == DIRECTION::YOrthogonal) {
+        s.mm = f.ynext(-2)[i.template minus<2, direction>()];
```

Comment:
This presumably is an issue in next as well where we just use one guard cell? I guess we might want (and I think I recall this has been suggested) that ShiftedMetric throws if stagger grids is enabled for the time being?

Reply:
I proposed to throw from `ShiftedMetric` if staggered grids were enabled before, but now I'd rather not. Soon `ShiftedMetric` should move into `Coordinates` and get a location; then it should work with staggering in x- and z-directions should work, I think. Also soon, I'm hoping to be able to use `ShiftedMetric::toFieldAligned` to get the inputs to staggered y-derivatives (might need to add some option/flag to turn off `calcYUpDown()` for staggered simulations, but I think that'd be the only change needed and would avoid having to introduce a whole new `ParallelTransform`).

Reply:
After some discussion with @d7919 we think we should be able to do staggered y-derivatives, but the result will be field-aligned, and need `fromFieldAligned` calling on it. We might even be able to do this for FCI as it might just be an interpolation.

There's more work to be done there though, and it might be easier once the y-tagging goes in so I'm going to leave it for another PR.